src/test/java/mate/academy/controller/BookControllerTest.java

```diff
@@ -0,0 +1,338 @@
+    }
+
+    @SneakyThrows
+    private static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("/database/categories/remove-all-categories.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("/database/books/remove-all-books.sql")
+            );
+        }
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Create a new book")
+    @Sql(
+            scripts = {
+                    "classpath:/database/books_categories/remove-all-books-categories.sql",
+                    "classpath:/database/books/remove-all-books.sql",
+            },
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(
+            scripts = {
+                    "classpath:/database/books_categories/remove-all-books-categories.sql",
+                    "classpath:/database/books/remove-all-books.sql",
+                    "classpath:/database/books/add-three-default-books.sql"
+            },
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    public void createBook_ValidRequest_Success() throws Exception {
+        // Given
+        CreateBookRequestDto requestDto = new CreateBookRequestDto()
+                .setTitle("Title")
+                .setAuthor("Author")
+                .setIsbn("1234567890")
+                .setPrice(BigDecimal.valueOf(9.99))
+                .setDescription("Description")
+                .setCoverImage("CoverImage")
+                .setCategoryIds(Set.of(1L));
+        BookDto expected = new BookDto()
+                .setTitle(requestDto.getTitle())
+                .setAuthor(requestDto.getAuthor())
+                .setIsbn(requestDto.getIsbn())
+                .setPrice(requestDto.getPrice())
+                .setDescription(requestDto.getDescription())
+                .setCoverImage(requestDto.getCoverImage())
+                .setCategoryIds(requestDto.getCategoryIds());
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        // When
+        MvcResult result = mockMvc.perform(
+                        post("/books")
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .accept(MediaType.APPLICATION_JSON)
+                ).andExpect(status().isCreated())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+
+        // Then
+        BookDto actual =
+                objectMapper.readValue(result.getResponse().getContentAsString(), BookDto.class);
+
+        EqualsBuilder.reflectionEquals(expected, actual, "id");

```

Comment:
alwaye use assertTrue with reflectionEquals
check other places
```suggestion
        assertTrue(EqualsBuilder.reflectionEquals(expected, actual, "id"));
```